### ISSUE TITLE
[Tests] Improve integration testing flow

### ISSUE THREAD

**Issue by verbanicm** [abcxyz/guardian#261]
### TL;DR

The current process requires a change to a Terraform resource to generate a change. Additionally, the apply happens after merge, which is ideal in the production workflow but not for testing.

The following changes should be done:

1) Remove all `test-*.yaml` workflows, except `test-plan.yml`
2) Rename `test-plan.yml` to `integration.yml`
3) Update Terraform resources in the `terraform` fodler to have a local backend and "empty" resources. We want to keep multiple folder to ensure the directories command works properly.

> [!NOTE]
> We aren't testing terraform, so these dont need to be "real" resources e.g.
> 
> ```terraform
> terraform {
>   backend "local" {}
> }
> 
> resource "null_resource" "empty" {}
> ```

4) Add an `apply` job after `plan_success` to ensure apply works
5) Add a `destroy` job after `apply` to ensure `guardian run` works
